File: lib/205.isomorphic_strings.dart
void main() {
  print(isIsomorphic("egg", "add"));
  print(isIsomorphic("foo", "bar"));
  print(isIsomorphic("paper", "title"));
  print(isIsomorphic("badc", "baba"));
}

bool isIsomorphic(String s, String t) {
  final map = <String, String>{};
  final map2 = <String, String>{};

  for (var i = 0; i < s.length; i++) {
    final value = map[s[i]];
    final value2 = map2[t[i]];

    if (value == null) {
      map[s[i]] = t[i];
    } else if (value != t[i]) {
      return false;
    }

    if (value2 == null) {
      map2[t[i]] = s[i];
    } else if (value2 != s[i]) {
      return false;
    }
  }

  return true;
}

File: src/205.isomorphic_strings.kt
fun main() {
    println(isIsomorphic("egg", "add"))
    println(isIsomorphic("foo", "bar"))
    println(isIsomorphic("paper", "title"))
    println(isIsomorphic("badc", "baba"))
}

fun isIsomorphic(s: String, t: String): Boolean {
    val map = mutableMapOf<Char, Char>()
    val map2 = mutableMapOf<Char, Char>()

    for (i in s.indices) {
        val value = map[s[i]]
        val value2 = map2[t[i]]

        if (value == null) {
            map[s[i]] = t[i]
        } else if (value != t[i]) {
            return false
        }

        if (value2 == null) {
            map2[t[i]] = s[i]
        } else if (value2 != s[i]) {
            return false
        }
    }

    return true
}